docset/winserver2025-ps/VMDirectStorage/Add-VMDirectVirtualDisk.md

```diff
@@ -1,55 +1,63 @@
 ---
 external help file: VMDirectStorage-help.xml
 Module Name: VMDirectStorage
-ms.date: 02/21/2024
+ms.date: 3/20/2025
 online version: https://learn.microsoft.com/powershell/module/vmdirectstorage/add-vmdirectvirtualdisk?view=windowsserver2025-ps&wt.mc_id=ps-gethelp
 schema: 2.0.0
 title: Add-VMDirectVirtualDisk
+ai-usage: ai-generated
 ---
 
 # Add-VMDirectVirtualDisk
 
 ## SYNOPSIS
-{{ Fill in the Synopsis }}
+
+Adds a direct-attached virtual disk to a Hyper-V virtual machine.
 
 ## SYNTAX
 
 ### ByVMName
+
 ```
 Add-VMDirectVirtualDisk [-VMName] <String[]> [-CimSession <CimSession[]>] [[-ControllerType] <ControllerType>]
  [[-ControllerNumber] <Int32>] [-ControllerLocation <Int32>] [-VirtualDiskUniqueId <String>]
  [<CommonParameters>]
 ```
 
 ### ByVM
+
 ```
 Add-VMDirectVirtualDisk [-VM] <VirtualMachine[]> [[-ControllerType] <ControllerType>]
  [[-ControllerNumber] <Int32>] [-ControllerLocation <Int32>] [-VirtualDiskUniqueId <String>]
  [<CommonParameters>]
 ```
 
 ### ByVMDriveController
+
 ```
 Add-VMDirectVirtualDisk [-VMDriveController] <VMDriveController> [-ControllerLocation <Int32>]
  [-VirtualDiskUniqueId <String>] [<CommonParameters>]
 ```
 
 ## DESCRIPTION
-{{ Fill in the Description }}
+
+The `Add-VMDirectVirtualDisk` cmdlet attaches an existing virtual disk directly to a Hyper-V virtual machine. Direct-attached virtual disks provide improved storage performance by bypassing traditional virtual storage layers.
 
 ## EXAMPLES
 
 ### Example 1
+
 ```powershell
-PS C:\> {{ Add example code here }}
+PS C:\> Add-VMDirectVirtualDisk -VMName "VM01" -ControllerType SCSI -ControllerNumber 0 -ControllerLocation 1 -VirtualDiskUniqueId "12345678-ABCD-1234-ABCD-1234567890AB"
```

Comment:
Don't use the prompt in the example. For long command lines, use PowerShell splatting.
```suggestion
$parameters = @{
    VMName              = "VM01"
    ControllerType      = 'SCSI'
    ControllerNumber    = 0
    ControllerLocation  = 1
    VirtualDiskUniqueId = "12345678-ABCD-1234-ABCD-1234567890AB"
}
Add-VMDirectVirtualDisk @parameters
```